### PR TITLE
Truly unmodifiable set

### PR DESCRIPTION
**Pull request by ggam**
The original Set was still modifiable from the outside

### CODE REVIEW

File: impl/src/main/java/org/glassfish/soteria/servlet/AuthenticationData.java [hunk 1]
```diff
@@ -40,7 +41,7 @@
     
     public AuthenticationData(Principal principal, Set<String> groups) {
         this.principal = principal;
-        this.groups =  unmodifiableSet(groups);
+        this.groups = new HashSet<>(unmodifiableSet(groups));
```
Comment (arjantijms) [inline impl/src/main/java/org/glassfish/soteria/servlet/AuthenticationData.java]: What about either:

```java
new HashSet<>(groups);
```

or

```java
unmodifiableSet(new HashSet<>(groups));
```

The first makes a defensive copy, while the second makes a defensive copy and signals to the user it's useless to modify it.

Making a copy of an unmodifiable set doesn't seem to do anything ;)

Comment (ggam) [reply impl/src/main/java/org/glassfish/soteria/servlet/AuthenticationData.java]: Ouch! It was too easy to make it wrong. ``unmodifiableSet(new HashSet<>(groups));` was really what I had in mind. I'll fix it.

**Comment by ggam** [comment]
Fixed.

**Comment by arjantijms** [comment]
That looks better, thx! ;)

**Pull request merged**